utils/src/main/scala/za/co/absa/enceladus/utils/performance/PerformanceMetricTools.scala

```diff
@@ -94,12 +95,13 @@ object PerformanceMetricTools {
 
   /** Returns the number of records failed, the number of records succeeded and the total number of errors encountered
     * when running a Standardization or a Dynamic Conformance job. */
-  private def getNumberOfErrors(spark: SparkSession, outputPath: String): (Long, Long, Long) = {
+  private def getNumberOfErrors(spark: SparkSession, outputPath: String): (Long, Long, Long, Long) = {
     val df = spark.read.parquet(outputPath)
     val errorCountColumn = SchemaUtils.getClosestUniqueName("enceladus_error_count", df.schema)
     val errCol = col(ErrorMessage.errorColumnName)
     val numRecordsFailed = df.filter(size(errCol) > 0).count
     val numRecordsSuccessful = df.filter(size(errCol) === 0).count
+    val numTotal = df.count
```

Comment:
As far as I understand, the 3 lines above are independent actions, which will cause Spark to traverse the RDD 3 times. Is it possible to combine them all into one action?

Reply:
Even if not pared down to a single action, isn't `numTotal == numRecordsFailed + numRecordsSuccessful `?

Reply:
Right now `numTotal == numRecordsFailed + numRecordsSuccessful` holds. But it might change the meaning according to user's request.

But after talking to Dima we concluded if users want to change a meaning of a metric a new metric should be added instead. So will combine calculate it like you said.

Reply:
Maybe something like? 
(numRecordsFailed,  numRecordsSuccessful) = df
.map( row => (size(row(errCol)) > 0 ? (1,0) : (0,1) )
.reduce( (a,b) => (a._1 + b._1, a._2+b._2 ) )

Reply:
The approach you specified does not compile. Tried a different one - with 2 aggregates.
```scala
    val result = df
      .agg(sum( when(size(errCol)>0, 1).otherwise(0) ).as("failed"),
        sum( when(size(errCol)>0, 0).otherwise(1) ).as("succeeded") )
      .collect()(0)

    val numRecordsFailed = result(0)
    val numRecordsSuccessful = result(1)
```
It works, but has caveats. Need to chcek against nulls if there are no records.

I'd prefer leaving this piece of code like it was for now.